File: qs-hadoop-userLog-scala/src/main/scala/com/qs/log/utils/MYSQLUtils.scala
package com.qs.log.utils

import java.sql.{Connection, DriverManager, PreparedStatement}


/**
  * Mysql 工具类
  */
object MYSQLUtils {

  val url = "jdbc:mysql://192.168.1.197:3306/hadoop?characterEncoding=utf-8"
  val userName = "dev"
  val password = "dev"



  def getConnection : Connection = {
    DriverManager.getConnection(url,userName,password)
  }


  def release(preparedStatement: PreparedStatement,connection: Connection):Unit = {
    try {
      if (preparedStatement != null) {
        preparedStatement.close()
      }
    } catch {
      case exception: Exception => exception.printStackTrace()
    }finally {
      if (connection != null) {
        connection.close()
      }
    }

  }

  def main(args: Array[String]): Unit = {
    println(getConnection)
  }

}

File: qs-hadoop-webUi/src/main/java/com/qs/hadoop/utils/MYSQLUtils.java
package com.qs.hadoop.utils;

import java.sql.*;

/**
 * mysql 工具类
 */
public class MYSQLUtils {

    private static final String url = "jdbc:mysql://192.168.1.197:3306/hadoop?characterEncoding=utf-8";
    private static final String userName = "dev";
    private static final String password = "dev";
    private static final String driver = "com.mysql.jdbc.Driver";


    public static Connection getConnection() {
        Connection connection = null;
        try {
            Class.forName(driver);
            connection =  DriverManager.getConnection(url, userName, password);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return connection;
    }


    public static void release(Connection connection, PreparedStatement preparedStatement, ResultSet resultSet) {
        try {
            if (resultSet != null) {
                resultSet.close();
            }
            if (preparedStatement != null) {
                preparedStatement.close();
            }
            if (connection != null) {
                connection.close();
            }
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public static void main(String[] args) {
        System.out.println("getConnection() = " + getConnection());
    }
}
